pkg/alerts/alertmanager/client.go

```diff
@@ -0,0 +1,121 @@
+package alertmanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+)
+
+const (
+	alertManagerPathPrefix = "/api/v2"
+	// the default context timeout for alert manager client
+	// feel free to change this value/set a corresponding env var if needed
+	defaultTimeOut = 10 * time.Second
+)
+
+// New is a constructor of AlertManagerClient
+//
+// if no url flag is given => error
+func New(alertManagerURL, alertManagerToken string) (*Client, error) {
+	if alertManagerURL == "" {
+		return nil, fmt.Errorf("no alert manager url found")
+	}
+	return &Client{
+		Token:   alertManagerToken,
+		HostURL: alertManagerURL,
+		Client:  new(http.Client),
+	}, nil
+}
+
+// Status builds the Status endpoint
+func (c *Client) Status() *StatusEndpoint {
+	return &StatusEndpoint{
+		Client: *c,
+	}
+}
+
+// Silences builds the Silences endpoint
+func (c *Client) Silences() *SilencesEndpoint {
+	return &SilencesEndpoint{
+		Client: *c,
+	}
+}
+
+// BuildURL builds the full URL for Status Endpoint
+func (s *StatusEndpoint) BuildURL() error {
+	url, err := url.Parse(s.HostURL)
+	if err != nil {
+		return err
+	}
+	url.Path = path.Join(alertManagerPathPrefix, "status")
+	s.FullURL = url.String()
+	return nil
+}
+
+// Get receives information about alert manager overall status
+func (s *StatusEndpoint) Get() (*StatusResponse, error) {
+	err := s.BuildURL()
```

Comment:
Why does this BuildURL() method exist, while the StatusEndpoint data structure receives the FullURL (which I suppose contains `status`)? 

If FullURL does not intend to be the complete URL (status included) and only intends to be base URL, then I guess having this logic could make sense.

However, in that case, I think I would implement it using the standard go net/url instead, because we have a recent enough version of go. This allows us to remove BuildURL() completely.

For example, in statusendpoint get(), I would do url.JoinPath(s.baseUrl,"status") (see https://pkg.go.dev/net/url@go1.20#JoinPath).

This automatically parses the url and will output any errors of parsing: https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/net/url/url.go;l=1262.

Hence the purpose of L61 of this file is still a one liner, yet it removes the whole BuildURL.

On top of that, I feel it is easier to grasp the Get() if you have all the details there.